docs/berkeley-upgrade/migrating-archive-database-to-berkeley.mdx

```diff
@@ -29,67 +29,181 @@ You can choose packages as you wish. The following commands help you install Deb
+After initial migration we have migrated data till last canonical block. However mainnet data is progressing with new blocks 
+which we need to migrate again and again until fork block is announced 
 
-   docker cp $id:/etc/mina/rosetta/archive/zkapp_tables.sql - > zkapp_tables.sql
 
-   docker rm -v $id
-   ```
+- Inputs
+   - Newest mainnet database
+   - Mainnet genesis ledger
+   - Replayer checkpoint from last run 
+   - Migrated berkeley schema from initial migration 
 
-### Migration Phase 1: Berkeley migration app run
+- Outputs
+   - Migrated mainnet database to berkeley from up to last canonical block
+   - Replayer checkpoint which can be used for next incremental migration
 
-Run the provided Berkeley migration application:
+##### Phase 1: Berkeley migration app run
 
 ```
-mina-berkeley-migration --batch-size 100 --config-file genesis_ledgers/mainnet.json
+mina-berkeley-migration --batch-size 1000 --config-file genesis_ledgers/mainnet.json
 --mainnet-archive-uri {mainnet_connection_string} --migrated-archive-uri {migrated_connection_string} --mainnet-blocks-bucket 
-{bucket name}
+{bucket name} --network "mainnet"
 ```
+where: 
+`--network` - precomputed blocks prefix 
 
-### Migration Phase 2: Replayer in migration mode runs
 
-Run the provided replayer app in migration mode:
+##### Phase 2: Replayer in migration mode run
 
-1. Prepare replayer config file.
+```
+ mina-replayer --migration-mode --archive-uri {migrated_connection_string}  --input-file replayer-checkpoint-XXXXX.json --checkpoint-interval 10000
+```
 
-   Replayer config must contain the Mainnet ledger as starting point:
+where `replayer-checkpoint-XXXXX.json` - is last checkpoint generated from previous migration
 
-   ```
-   jq '.ledger.accounts' mainnet.json | jq  '{genesis_ledger: {accounts: .}}' > replayer_input_config.json"
-   ```
 
-2. Run the import in migration mode:
+Incremental migration can be run continuously on top initial migration or last incremental until fork block is anounced 
 
-   ```
-   mina-replayer --migration-mode --archive-uri {migrated_connection_string}  --input-file replayer_input_config.json --checkpoint-interval 100 --output-file  replayer_output.json 
-   ```
+#### Stage 3: Remainder migration
+
+When fork block is announced we can tackle the remainder migration this is the last migration run 
+we would want to perform.
+
+- Inputs
+   - Newest mainnet database
+   - Mainnet genesis ledger
+   - Replayer checkpoint from last run 
+   - Migrated berkeley schema from last run
+   - Fork block state hash
+
+- Outputs
+   - Migrated mainnet database to berkeley from up fork point
+   - Replayer checkpoint which can be used for replayer run on berkeley schema
+
+
+##### Phase 1: Berkeley migration app run
+
+Noticed decrease in `batch-size` for precomputed blocks as tool may hit missing precomputed blocks problem if we would like 
+to fetch too many of them in single batch while trying to migrate newest blocks in the chain
```

Comment:
```suggestion
You might notice a decrease in `batch-size` for precomputed blocks as the tool might hit missing precomputed blocks problem if you fetch too many of them in a single batch while trying to migrate the newest blocks in the chain:
```

Reply:
@dkijania this sentence introduces a command, so I guessed that a user might notice a decreased batch size, maybe instead it needs to say "to reduce the batch-size?"